### PR TITLE
DOCS-#3015: Document `modin.experimental.sklearn` module

### LINKED ISSUE #3015: Document modin.experimental.sklearn
This should include docstrings and purposes of everything inside modin.experimental.sklearn.

### PR DESCRIPTION
**Pull request by prutskov**
Signed-off-by: Alexey Prutskov <alexey.prutskov@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3015 ? 
- [ ] tests added and passing

### CODE REVIEW

**Comment by prutskov** [comment]
Built docs can be found here: https://modin--3016.org.readthedocs.build/en/3016/index.html

**Review by vnlitvinov — COMMENTED** [review]
Minor notes only

File: modin/experimental/sklearn/model_selection/train_test_split.py [hunk 1]
```diff
@@ -11,8 +11,26 @@
...
+    """
+    Split input data to train and test data.
+
+    Parameters
+    ----------
+    df : modin.pandas.DataFrame / modin.pandas.Series
+        Data to split.
+    **options : dict
+        Keyword arguments. If `train_size` key isn't provided
+        `train_size` will be 0.75.
```
[the remark this answers is not in the document]
Comment (prutskov) [reply modin/experimental/sklearn/model_selection/train_test_split.py:27]: Do you mean we need to add `FIXME` about changing parameter `**options`->`train_size=0.75`?

File: docs/flow/modin/experimental/sklearn.rst [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Scikit-learn module description
+"""""""""""""""""""""""""""""""
+
+This module holds experimental scikit-learn specific functionality for Modin.
```
[the remark this answers is not in the document]
Comment (prutskov) [reply docs/flow/modin/experimental/sklearn.rst]: fixed

File: modin/experimental/sklearn/model_selection/train_test_split.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vnlitvinov) [reply modin/experimental/sklearn/model_selection/train_test_split.py:27]: yes

[the remark this answers is not in the document]
Comment (prutskov) [reply modin/experimental/sklearn/model_selection/train_test_split.py:28]: added

**Review by vnlitvinov — APPROVED** [review]
LGTM

File: docs/flow/modin/experimental/index.rst [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Experimental
+============
+
+Experimental part holds experimental functionality that is under development right now
```
Comment (devin-petersohn) [inline docs/flow/modin/experimental/index.rst:4]:
```suggestion
``modin.experimental`` holds experimental functionality that is under development right now
```

**Review by devin-petersohn — CHANGES_REQUESTED** [review]
Just one minor wording change for grammar. Thanks!

File: docs/flow/modin/experimental/index.rst (hunk 3 above)
Comment (prutskov) [reply docs/flow/modin/experimental/index.rst]: fixed

**Review by devin-petersohn — APPROVED** [review]
Looks good @prutskov!

**Pull request merged**